docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+# Writing Console Apps: A Step by Step Guide
+
+This guide will show you how to use the .NET CLI tooling to build cross-platform console apps incrementally.  It will start with the most basic console app and eventually span multiple projects, including testing.
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
+
+## Prerequisites
+
+Before you begin, ensure you have the [latest .NET CLI tooling](http://dotnet.github.io/getting-started/).  You'll also need a text editor.
+
+## Hello, Console App!
+
+First, navigate to or create a new folder with a name you like.  "Hello" is the name chosen for the sample code, which can be found [here](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/Hello).
+
+Open up a command prompt and type the following:
+
+```
+$ dotnet new
+$ dotnet restore
+$ dotnet compile
+```
+
+Let's do a quick walkthrough:
+
+1. `$ dotnet init`
+
+   `dotnet init` created an up-to-date `project.json` file with NuGet dependencies necessary to build a console app.  It also created a `Program.cs`, a basic file containing the entry point for the application.
+   
+   `project.json`:
+   ```javascript
+   {
+        "version": "1.0.0-*",
+        "compilationOptions": {
+            "emitEntryPoint": true
+        },
+    
+        "dependencies": {
```

Comment:
The only dependency nowadays is STdLib as you can see in the `dotnet new` output.